### DISCUSSION TITLE
React Native example

### DISCUSSION THREAD

**Discussion by martinnormark** [observablehq/plot#1108] [Q&A]
Anyone seen an example of Plot with React Native? Rendering is one thing, but what about gestures for highlighting, showing tooltips etc?

**Comment by harshagarwalnyu** [comment]
Observable Plot renders to SVG/HTML DOM, which React Native does not provide natively. There are a few viable paths depending on your needs:

**Option 1 — WebView (simplest, full Plot feature set)**
```jsx
import { WebView } from 'react-native-webview';

const html = `
  <html><body>
  <script type="module">
    import * as Plot from "https://cdn.jsdelivr.net/npm/@observablehq/plot/+esm";
    document.body.appendChild(Plot.plot({ marks: [...] }));
  </script>
  </body></html>
`;

<WebView source={{ html }} style={{ width: 400, height: 300 }} />
```
Gestures and tooltips work because the full browser engine runs inside the WebView. Use `postMessage`/`onMessage` to pass data in and events out. Main drawback: WebView startup cost, no native look.

**Option 2 — react-native-svg (native rendering, manual port)**
Render Plot's output marks manually using `react-native-svg`. You'd replace Plot's SVG output with equivalent `<Svg>`, `<Line>`, `<Circle>` etc. components. Gestures use `react-native-gesture-handler`.

This is significant work — Plot does a lot of layout math — but gives fully native rendering and gesture integration.

**Option 3 — Victory Native (drop-in alternative)**
[Victory Native XL](https://github.com/FormidableLabs/victory-native-xl) is built on react-native-svg and Skia. It handles gestures, tooltips, and animations natively with an API closer to Recharts than Plot. If you don't specifically need Plot's mark composition model, this is the pragmatic choice for RN.

```bash
npm install victory-native @shopify/react-native-skia react-native-reanimated react-native-gesture-handler
```

For gestures + tooltips specifically: Victory Native XL has `useChartPressState()` that tracks finger position and returns interpolated data values — the exact equivalent of Plot's `tip` mark pointer tracking.
